src/main/java/data/Event.java

```diff
@@ -0,0 +1,47 @@
+package main.java.data;
+
+import main.java.common.Message;
+import main.java.exception.ChatbotException;
+
+import java.time.LocalDate;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Event extends Task {
+
+    public Event(String description, boolean isDone, LocalDate timestamp) {
+        super(description, "E", isDone, timestamp);
+    }
+
+    public static Event newEvent(String raw) throws ChatbotException {
+        if (raw.length() == 0) {
+            throw new ChatbotException("Oh boy your event is empty.");
+        }
```

Comment:
I like your use of a guard clause here! It makes the happy path much more clear.